src/compute/src/lib.rs

```diff
@@ -133,6 +134,28 @@ pub struct ComputeNodeOpts {
     pub heap_profiling_dir: Option<String>,
 }
 
+impl ComputeNodeOpts {
+    pub fn new_for_single_node() -> Self {
```

Comment:
Slightly prefer to put all these configs in `src/cmd_all/src/single_node.rs`, instead of each Opt struct's file.

Reply:
Possible I guess. We need to expose a `new` constructor in each file.
Then in `single_node` construct them.

Reply:
We can just make all fields pub